src/sources/util/http.rs

```diff
@@ -202,11 +202,19 @@ pub trait HttpSource: Clone + Send + Sync + 'static {
             for s in path.split('/').filter(|&x| !x.is_empty()) {
                 filter = filter.and(warp::path(s.to_string())).boxed()
             }
-            if strict_path {
-                debug!(message = "Strict path enabled.");
-                filter = filter.and(warp::path::end()).boxed();
-            }
             let svc = filter
+                .and(warp::path::tail())
+                .and_then(move |tail: Tail| async move {
+                    if !strict_path || tail.as_str().is_empty() {
+                        Ok(())
+                    } else {
+                        debug!(message = "Path rejected.");
+                        Err(warp::reject::custom(
+                            ErrorMessage::new(StatusCode::NOT_FOUND,
+                            StatusCode::NOT_FOUND.canonical_reason().unwrap().to_string())
```

Comment:
Could we use `unwrap_or_default()` here? I realize that this particular code always has a canonical reason, but `unwrap()`s tend to jump out.

Reply:
That sounds safer. In this case I think we may directly go for `"Not found".to_string()` WDYT ?

Reply:
I'm also fine with that 😄

Reply:
It looks like there is also a `warp::reject::not_found()` you could use instead of `warp::reject::custom()`:

https://docs.rs/warp/0.2.5/warp/reject/fn.not_found.html